client/src/utils/groups.ts

```diff
@@ -11,25 +11,59 @@ export function aggregateGroups(groups: Array<Group>) {
         security: [],
         mailEnabled: [],
         others: [],
+        teamConnected: [],
+        public: [],
+        private: [],
+        orphaned: [],
+        active: []
     }
     for (const group of groups) {
-        const key = getGroupAggregationKey(group)
-        aggregatedGroups[key].push(group)
+        const keys = getGroupAggregationKey(group)
+        keys.forEach(key => {
+            aggregatedGroups[key].push(group)
+        })
+        // aggregatedGroups[key].push(group)
     }
 
     return aggregatedGroups
 }
 
-function getGroupAggregationKey(group: Group): GroupAggKey {
+function getGroupAggregationKey(group: Group): GroupAggKey[] {
+    const outputKeys: GroupAggKey[] = [];
     if (group.groupTypes.includes('Unified')) {
-        return 'unified'
+        outputKeys.push('unified')
     }
     if (!group.mailEnabled && group.securityEnabled) {
-        return group.groupTypes.includes('DynamicMembership') ? 'dynamic' : 'security'
+        const key = group.groupTypes.includes('DynamicMembership') ? 'dynamic' : 'security'
+        outputKeys.push(key)
     }
     if (group.mailEnabled) {
-        if (!group.securityEnabled) return 'distribution'
-        return 'mailEnabled'
+        if (!group.securityEnabled) outputKeys.push('distribution')
+        outputKeys.push('mailEnabled')
     }
-    return 'others'
+    if(group.resourceProvisioningOptions.some(option => option === "Team")){
+        outputKeys.push("teamConnected")
+    }
+
+    if(group.visibility === "Public"){
+        outputKeys.push("public")
+    }
+
+    if(group.visibility === "Private"){
+        outputKeys.push("private")
+    }
```

Comment:
This is getting pretty ugly, we probably want to rewrite this and avoid keep using if